src/ert/run_models/event.py

```diff
@@ -56,3 +69,88 @@ class RunModelErrorEvent(RunModelEvent):
     def write_as_csv(self, output_path: Path | None) -> None:
         if output_path and self.data:
             self.data.to_csv("Report", output_path / str(self.run_id))
+
+
+StatusEvents = (
+    AnalysisEvent
+    | AnalysisStatusEvent
+    | AnalysisTimeEvent
+    | EndEvent
+    | FullSnapshotEvent
+    | SnapshotUpdateEvent
+    | RunModelErrorEvent
+    | RunModelStatusEvent
+    | RunModelTimeEvent
+    | RunModelUpdateBeginEvent
+    | RunModelDataEvent
+    | RunModelUpdateEndEvent
+)
+
+
+EVENT_MAPPING = {
+    "AnalysisEvent": AnalysisEvent,
+    "AnalysisStatusEvent": AnalysisStatusEvent,
+    "AnalysisTimeEvent": AnalysisTimeEvent,
+    "EndEvent": EndEvent,
+    "FullSnapshotEvent": FullSnapshotEvent,
+    "SnapshotUpdateEvent": SnapshotUpdateEvent,
+    "RunModelErrorEvent": RunModelErrorEvent,
+    "RunModelStatusEvent": RunModelStatusEvent,
+    "RunModelTimeEvent": RunModelTimeEvent,
+    "RunModelUpdateBeginEvent": RunModelUpdateBeginEvent,
+    "RunModelDataEvent": RunModelDataEvent,
+    "RunModelUpdateEndEvent": RunModelUpdateEndEvent,
+}
+
+
+def status_event_from_json(json_str: str) -> StatusEvents:
+    json_dict = json.loads(json_str)
+    event_type = json_dict.pop("event_type", None)
+
+    match event_type:
+        case FullSnapshotEvent.__name__:
+            snapshot = EnsembleSnapshot.from_nested_dict(json_dict["snapshot"])
+            json_dict["snapshot"] = snapshot
+            return FullSnapshotEvent(**json_dict)
+        case SnapshotUpdateEvent.__name__:
+            snapshot = EnsembleSnapshot.from_nested_dict(json_dict["snapshot"])
+            json_dict["snapshot"] = snapshot
+            return SnapshotUpdateEvent(**json_dict)
+        case RunModelDataEvent.__name__ | RunModelUpdateEndEvent.__name__:
+            if "run_id" in json_dict and isinstance(json_dict["run_id"], str):
+                json_dict["run_id"] = UUID(json_dict["run_id"])
+            if json_dict.get("data"):
+                json_dict["data"] = DataSection(**json_dict["data"])
+            return EVENT_MAPPING[event_type](**json_dict)
+        case _:
+            if event_type in EVENT_MAPPING:
+                if "run_id" in json_dict and isinstance(json_dict["run_id"], str):
+                    json_dict["run_id"] = UUID(json_dict["run_id"])
+                return EVENT_MAPPING[event_type](**json_dict)
+            else:
+                raise TypeError(f"Unknown status event type {event_type}")
+
+
+def status_event_to_json(event: StatusEvents) -> str:
+    match event:
+        case FullSnapshotEvent() | SnapshotUpdateEvent():
+            assert event.snapshot is not None
+            event_dict = asdict(event)
+            event_dict.update(
+                {
+                    "snapshot": event.snapshot.to_dict(),
+                    "event_type": event.__class__.__name__,
```

Comment:
Might be better to add this directly to the events? 
```python
event_type: Literal['FullSnapshotEvent'] = 'FullSnapshotEvent'
```
the pro is it is more explicit, but on the flip side it is more verbose and has to be kept in sync for all events 🤔

Reply:
That is a good idea the effort to keep it all in sync should not be too much.